qt_ui/windows/QLiberationWindow.py

```diff
@@ -367,6 +371,32 @@ def onGameGenerated(self, game: Game):
         self.game = game
         GameUpdateSignal.get_instance().game_loaded.emit(self.game)
 
+    def onEndGame(self, state: TurnState):
+        result = None
+        if state == TurnState.WIN:
+            result = QMessageBox.information(
+                QApplication.focusWidget(),
+                "Victory!",
+                "You have won the campaign, do you wish to start a new one?",
+                QMessageBox.Yes,
+                QMessageBox.No,
+            )
+        elif state == TurnState.LOSS:
+            result = QMessageBox.information(
+                QApplication.focusWidget(),
+                "Defeat!",
+                "You have lost the campaign, do you wish to start a new one?",
+                QMessageBox.Yes,
+                QMessageBox.No,
+            )
+        for window in QApplication.topLevelWidgets():
+            if window is not self:
+                window.close()
+        sleep(0.5)
```

Comment:
This is needed to let the map fully update <ins>after the last turn</ins> since the next line will "release" the game. If that happens too soon, the map doesn't update properly and will still depict the image from the last turn, i.e. <ins>before the last turn</ins>. It's specifically the case with a cheat-capture, less applicable to winning/losing through the debriefing window, i.e. winning/losing after actually flying the "last mission".